airbyte-connector-builder-server/connector_builder/impl/default_api.py

```diff
@@ -2,21 +2,344 @@
+        except ValidationError as error:
+            return KnownExceptionInfo(
+                message=f"Invalid connector manifest with error: {error.message}", exception_class_name=ValidationError.__name__
+            )
+
+        single_slice = StreamReadSlices(pages=[])
+        log_messages = []
+
+        first_page = True
+        current_records: List[object] = []
+        current_page_request: Optional[HttpRequest] = None
+        current_page_response: Optional[HttpResponse] = None
+        try:
+            # Message groups are partitioned according to when request log messages are received. Subsequent response log messages
+            # and record messages belong to the prior request log message and when we encounter another request, append the latest
+            # message group
+            for message in adapter.read_stream(stream_read_request_body.stream, stream_read_request_body.config):
+                if first_page and message.type == Type.LOG and message.log.message.startswith("request:"):
+                    first_page = False
+                    request = self._create_request_from_log_message(message.log)
+                    current_page_request = request
+                elif message.type == Type.LOG and message.log.message.startswith("request:"):
+                    if not current_page_request or not current_page_response:
+                        self.logger.warning("Every message grouping should have at least one request and response")
+                    single_slice.pages.append(
+                        StreamReadPages(request=current_page_request, response=current_page_response, records=current_records)
+                    )
+                    current_page_request = self._create_request_from_log_message(message.log)
+                    current_records = []
+                elif message.type == Type.LOG and message.log.message.startswith("response:"):
+                    current_page_response = self._create_response_from_log_message(message.log)
+                elif message.type == Type.LOG:
+                    log_messages.append({"message": message.log.message})
+                elif message.type == Type.RECORD:
+                    current_records.append(message.record.data)
+            else:
+                if not current_page_request or not current_page_response:
+                    self.logger.warning("Every message grouping should have at least one request and response")
+                single_slice.pages.append(
+                    StreamReadPages(request=current_page_request, response=current_page_response, records=current_records)
+                )
+        except KeyError as error:
+            return KnownExceptionInfo(message=error.args[0])
+
+        return StreamRead(logs=log_messages, slices=[single_slice])
+
+    def _create_request_from_log_message(self, log_message: AirbyteLogMessage) -> Optional[HttpRequest]:
```

Comment:
might be good to add a TODO acknowledging the hack being done here and linking the github issue that will fix this